includes/class-fs-plugin-updater.php

```diff
@@ -1078,13 +1108,56 @@ private function is_premium_plugin_active( $plugin_id = false ) {
          *
          * @return string|WP_Error
          */
-        function _maybe_adjust_source_dir( $source, $remote_source, $upgrader ) {
+        static function _maybe_adjust_source_dir( $source, $remote_source, $upgrader ) {

```

Comment:
I don't understand the reasoning converting this method to static. On the contrary, it looks like it make things more complex since you don't have a direct access to the `$this->_fs` object. Am I missing something?